packages/core/src/client.ts

```diff
@@ -418,4 +361,108 @@ export abstract class Client {
 
     return tags.toString().split(TAG_SEPARATOR).filter((tag) => NOT_BLANK.test(tag))
   }
+
+  private __runPreconditions(notice: Notice): Error | null | Promise<Error | null> {
+    let preConditionError: Error | null = null
+    if (!notice) {
+      this.logger.debug('failed to build error report')
+      preConditionError = new Error('failed to build error report')
+    }
+
+    if (this.config.reportData === false) {
+      this.logger.debug('skipping error report: honeybadger.js is disabled', notice)
+      preConditionError = new Error('honeybadger.js is disabled')
+    }
+
+    if (this.__developmentMode()) {
+      this.logger.log('honeybadger.js is in development mode; the following error report will be sent in production.', notice)
+      preConditionError = new Error('honeybadger.js is in development mode')
+    }
+
+    if (!this.config.apiKey) {
+      this.logger.warn('could not send error report: no API key has been configured', notice)
+      preConditionError = new Error('missing API key')
+    }
+
+    const beforeNotifyResult = runBeforeNotifyHandlers(notice, this.__beforeNotifyHandlers)
```

Comment:
If there's a `preConditionError`, why do we still run the `beforeNotifyHandlers`?